src/utils/parsing.ts

```diff
@@ -0,0 +1,74 @@
+function applyRegex(regex: RegExp, s: string): string | null {
+  const match = s.match(regex);
+  return match ? match[1] : null;
+}
+
+export const parseZ: (s: string) => string | null = applyRegex.bind(
+  null,
+  /Z(\d+)/
```

Comment:
Based on the spec in the document, I think these  regexs might be
`/(^|[_-])(z,slice)(\d+)/i`, `/(^|[_-])(t,time,daty,hour,min,wk,d,h)(\d+)/i`, `/(^|[_-])(xy,x,y,stage,pos)(\d+)/i`, `/(^|[_-])(ch,channel,wave,c)(.+?)([_-\.]|$)/i`

Reply:
See d65ef36.